app/models/statement.rb

```diff
@@ -14,6 +14,28 @@ class Statement < ApplicationRecord
 
   default_scope -> { order(created_at: :desc) }
 
+  scope :for_accounts, ->(accounts) { where(account_id: accounts) if accounts.present? }
+  scope :for_sent_to, lambda { |sent_to|
+    where(account: Account.joins(:notify_users).where(account_users: { user_id: sent_to })) if sent_to.present?
+  }
+
+  scope :created_between, lambda { |start_at, end_at|
+    if start_at
+      where(created_at: start_at..(end_at || DateTime::Infinity.new))
+    elsif end_at
+      where(arel_table[:created_at].lt(end_at))
+    end
+  }
+
+  RECONCILED_SQL = OrderDetail.unreconciled.where("order_details.statement_id = statements.id").select(1).to_sql
```

Comment:
@vandrijevik Alternate queries:

```ruby
scope :unreconciled, -> { where(id: OrderDetail.unreconciled.where.not(statement_id: nil).select(:statement_id)) }
  scope :reconciled, -> { where.not(id: OrderDetail.unreconciled.where.not(statement_id: nil).select(:statement_id)) }
```
I'm not sure if there's a really good way to dry that up.